## Assignment 2 - Segmenting and Clustering Neighborhoods in Toronto - Part 2

### Extracting the Latitude and Longitude of Canada Ontario

In [1]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values

# libraries for displaying images
from IPython.display import Image 
from IPython.core.display import HTML 
    
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize

!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library

print('Folium installed')
print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Solving environment: done

# All requested packages already installed.

Folium installed
Libraries imported.


In [2]:
import geocoder # import geocoder

# initialize your variable to None
lat_lng_coords = None

# loop until you get the coordinates
while(lat_lng_coords is None):
  g = geocoder.google('{}, Toronto, Ontario'.format(postal_code))
  lat_lng_coords = g.latlng

latitude = lat_lng_coords[0]
longitude = lat_lng_coords[1]


NameError: name 'postal_code' is not defined

#### Couldn't figure out the error, So I used the csv given in the link  

In [3]:
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner

link = "http://cocl.us/Geospatial_data"
df1 = pd.read_csv(link)

df1.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [4]:
df1.shape

(103, 3)

Both the data frames have 103 rows and 3 columns

#### Changing the column name Postal code to Postcode to merge the two data frames together

In [5]:
df1.columns = ['Postcode','Latitude','Longitude']

cols = df1.columns.tolist()
cols

['Postcode', 'Latitude', 'Longitude']

#### Read in the CSV file saved in the previous assignment

In [6]:
link = "https://raw.githubusercontent.com/Shekhar-rv/Coursera_Capstone/master/df_can.csv"
df = pd.read_csv(link,index_col=0)
df.head()

Postcode      Borough                           Neighbourhood
0      M1B  Scarborough                          Rouge ,Malvern
1      M1C  Scarborough  Highland Creek ,Rouge Hill ,Port Union
2      M1E  Scarborough       Guildwood ,Morningside ,West Hill
3      M1G  Scarborough                                  Woburn
4      M1H  Scarborough                               Cedarbrae

#### Merging the two data frames together based on their Postcode

In [7]:
df_new = pd.merge(df, df1, on='Postcode')
df_new.head(12)

Postcode      Borough                                      Neighbourhood  \
0       M1B  Scarborough                                     Rouge ,Malvern   
1       M1C  Scarborough             Highland Creek ,Rouge Hill ,Port Union   
2       M1E  Scarborough                  Guildwood ,Morningside ,West Hill   
3       M1G  Scarborough                                             Woburn   
4       M1H  Scarborough                                          Cedarbrae   
5       M1J  Scarborough                                Scarborough Village   
6       M1K  Scarborough        East Birchmount Park ,Ionview ,Kennedy Park   
7       M1L  Scarborough                    Clairlea ,Golden Mile ,Oakridge   
8       M1M  Scarborough    Cliffcrest ,Cliffside ,Scarborough Village West   
9       M1N  Scarborough                        Birch Cliff ,Cliffside West   
10      M1P  Scarborough  Dorset Park ,Scarborough Town Centre ,Wexford ...   
11      M1R  Scarborough                                  Maryvale ,Wexford   

     Latitude  Longitude  
0   43.806686 -79.194353  
1   43.784535 -79.160497  
2   43.763573 -79.188711  
3   43.770992 -79.216917  
4   43.773136 -79.239476  
5   43.744734 -79.239476  
6   43.727929 -79.262029  
7   43.711112 -79.284577  
8   43.716316 -79.239476  
9   43.692657 -79.264848  
10  43.757410 -79.273304  
11  43.750072 -79.295849

#### Save the file as a csv

In [8]:
df_new.to_csv(r'df_final.csv')

## End